vscode/src/extension.ts

```diff
@@ -754,20 +765,59 @@ class VsCodeExtension {
       modelConfig = await parseModelConfig(settingsPath);
     } catch (err) {
       this.state.logger.error("Error getting model config:", err);
+      this.state.modelProvider = undefined;
+      // Only dispose workflow if not fetching solution
```

Comment:
I'm not sure what we want to happen if the modelProvider changes while a solution is in progress, without this guard it basically stopped in the middle of a solution with no message which felt suboptimal, now the behavior should be that it will finish the in-progress solution but then update.